agent/agent.go

```diff
@@ -1217,6 +1196,33 @@ func (a *Agent) listenHTTP() ([]*HTTPServer, error) {
 	return servers, nil
 }
 
+// setupHTTPServer adds HTTP/2 support, ConnState, and a connection handshake timeout
+// to the http.Server.
+func setupHTTPServer(server *http.Server, connState func(net.Conn, http.ConnState), timeout time.Duration) error {
```

Comment:
This function was extracted primarily so that the HTTP2 setup can be tested. Ideally we would be able to test more of the HTTPServer setup, but that would require a much larger change to have all the constructors accept their dependencies instead of creating them.